source/UniverseObjects.cpp

```diff
@@ -442,6 +442,32 @@ void UniverseObjects::LoadFile(const string &path, bool debugMode)
 				categoryList.push_back(child.Token(0));
 			}
 		}
+		else if(key == "bunks")
+		{
+			for(const DataNode &child : node)
+			{
+				if(!child.Token(0).empty())
+				{
+					bool passengers = false;
+					bool crew = false;
+					if(child.Size() >= 2)
+					{
+						if(child.Token(1) == "passenger")
+							passengers = true;
+						else if(child.Token(1) == "crew")
+							crew = true;
+					}
+					if(child.Size() >= 3)
+					{
+						if(child.Token(2) == "passenger")
+							passengers = true;
+						else if(child.Token(2) == "crew")
+							crew = true;
+					}
```

Comment:
Not sure whats going on here with smth that's 3 or 2 tokens long without clear distinction between the two, but you can easily do this in one if; like this:
```suggestion
					int nameIndex = child.Size() >= 3 ? 2 : child.Size() >= 2 ? 1 : 0
					if(nameIndex)
					{
						if(child.Token(nameIndex) == "passenger")
							passengers = true;
						else if(child.Token(nameIndex) == "crew")
							crew = true;
					}
```

Reply:
Wont work because it only checks one of both like this

Reply:
Oh... I think I understood but it could use a small comment I think, like saying that the passenger and crew can be in any order; still is a better way to do this though I'm pretty sure

Reply:
I will look into it.

Reply:
Actually if the length is 3, that means we have both except if there was a typo?

Reply:
Hmm, I do not want to automatic add this attribute, just because someone added one letter at the end.

Reply:
Well this new logic would work as well, it's just a bit condensed